ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/ontologies/views/OntologiesViewImpl.java

```diff
@@ -0,0 +1,238 @@
+import com.google.gwt.event.logical.shared.SelectionHandler;
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.safehtml.shared.UriUtils;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiFactory;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.uibinder.client.UiHandler;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+import com.google.inject.assistedinject.Assisted;
+
+import com.sencha.gxt.cell.core.client.form.ComboBoxCell;
+import com.sencha.gxt.core.client.Style;
+import com.sencha.gxt.core.client.ValueProvider;
+import com.sencha.gxt.data.shared.LabelProvider;
+import com.sencha.gxt.data.shared.TreeStore;
+import com.sencha.gxt.widget.core.client.Composite;
+import com.sencha.gxt.widget.core.client.ContentPanel;
+import com.sencha.gxt.widget.core.client.button.TextButton;
+import com.sencha.gxt.widget.core.client.container.CardLayoutContainer;
+import com.sencha.gxt.widget.core.client.container.CenterLayoutContainer;
+import com.sencha.gxt.widget.core.client.event.SelectEvent;
+import com.sencha.gxt.widget.core.client.form.SimpleComboBox;
+import com.sencha.gxt.widget.core.client.selection.SelectionChangedEvent;
+import com.sencha.gxt.widget.core.client.tree.Tree;
+
+import java.util.List;
+
+/**
+ * @author aramsey
+ */
+public class OntologiesViewImpl extends Composite implements OntologiesView {
+
+    interface OntologiesViewImplUiBinder extends UiBinder<Widget, OntologiesViewImpl> {
+
+    }
+
+    private static OntologiesViewImplUiBinder uiBinder = GWT.create(OntologiesViewImplUiBinder.class);
+
+    @UiField TextButton addButton;
+    @UiField SimpleComboBox<Ontology> ontologyDropDown;
+    @UiField TextButton viewVersions;
+    @UiField TextButton saveHierarchy;
+    @UiField(provided = true) OntologiesViewAppearance appearance;
+    @UiField Tree<OntologyHierarchy, String> tree;
+    @UiField(provided = true) AppCategoriesView categoriesView;
+    @UiField(provided = true) AdminAppsGridView gridView;
+    @UiField CardLayoutContainer cards;
+    @UiField CenterLayoutContainer noTreePanel, emptyTreePanel;
+    @UiField ContentPanel treePanel;
+    @UiField TextButton publishButton;
+    @Inject EventBus eventBus;
+    @Inject DEClientConstants clientConstants;
+
+    private TreeStore<OntologyHierarchy> treeStore;
+    private Ontology activeOntology;
+
+    @Inject
+    public OntologiesViewImpl(OntologiesViewAppearance appearance,
+                              @Assisted TreeStore<OntologyHierarchy> treeStore,
+                              @Assisted AppCategoriesView categoriesView,
+                              @Assisted AdminAppsGridView gridView) {
+        this.appearance = appearance;
+        this.treeStore = treeStore;
+        this.categoriesView = categoriesView;
+        this.gridView = gridView;
+
+        initWidget(uiBinder.createAndBindUi(this));
+
+        publishButton.disable();
```

Comment:
Sure, this was a last minute thing I added when I realized I was clicking the buttons and nothing was happening because I hadn't selected an ontology yet. 😄 I figured I wouldn't be the only one to do that.